instructor/common.py

```diff
@@ -0,0 +1,604 @@
+                              context_instance=RequestContext(request))
+
+
+def assignments_edit_strains(request):
+    pk = request.session['assignment_id']
+    assignment = models.Assignment.objects.get(id=pk)
+    message = ''
+    extra_fields = 0
+    if 'add' in request.POST:
+        extra_fields = 1
+
+    StrainsFormSet = modelformset_factory(models.Strains, extra=extra_fields, fields=['name'])
+
+    if request.method == "POST":
+        formset = StrainsFormSet(request.POST)
+        formset.clean()
+
+        if formset.is_valid():
+            message = "Thank you"
+            entries = formset.save(commit=False)
+            for form in entries:
+                form.assignment = assignment
+                form.save()
+        else:
+            message = "Something went wrong"
+
+        if 'continue' in request.POST:
+            return redirect('common_assignments_edit_protocols')
+
+    formset = StrainsFormSet(queryset=models.Strains.objects.filter(assignment=assignment))
+    add_btn_num = formset.total_form_count()+1
+    return render_to_response('instructor/strains.html',
+                                  {'formset': formset,
+                                   'message': message,
+                                   'new':  request.session['new'],
+                                   'add_btn_num': add_btn_num,
+                                   'assignment': assignment
+                                  },
+                                  context_instance=RequestContext(request))
+
+
+def assignments_delete_strain(request):
+    models.Strains.objects.get(id=request.POST['pk']).delete()
+    return redirect('common_assignments_edit_strains')
+
+
+def assignments_edit_protocols(request):
+    pk = request.session['assignment_id']
+    assignment = models.Assignment.objects.get(id=pk)
+    message = ''
+    ProtocolsFormSet = modelformset_factory(models.Protocol, extra=1, can_delete=True, exclude=['assignment'])
+    if request.method == "POST":
+        formset = ProtocolsFormSet(request.POST)
+        formset.clean()
+        if ( formset.is_valid()):
+            message = "Thank you"
+            entries = formset.save(commit=False)
+            for form in entries:
+                form.assignment = assignment
+                form.save()
+        else:
+            message = "Something went wrong"
+
+    return render_to_response('instructor/protocols.html',
+                              {'formset': ProtocolsFormSet(
+                                  queryset=models.Protocol.objects.filter(assignment=assignment)),
+                               'message': message,
+                               'assignment': assignment
+                              },
+                              context_instance=RequestContext(request))
+
+
+def treatments_edit(request, assignment, protocol):
+    a = models.Assignment.objects.get(id=assignment)
+    p = models.Protocol.objects.get(id=protocol)
```

Comment:
exception handling, use more descriptive variable names please, i.e. `protocol` or at least `proto` and `assignment`